### ISSUE TITLE
HTML wrap length line cut-off is problematic

### ISSUE THREAD

**Issue by chenasraf** [beautify-web/js-beautify#837]
HTML cut off is longer than the character count specified in settings. Example:

I have this segment in a file. I set the wrap line length to 80 (same as my editor line) and the result after beautifying is this:

```
  <div autocomplete placeholder="Type to select a company or create a new one*" class="search-container search-filter-container company-search"
  ng-class="{'for-new-company': modal.newCompany}" search-model="modal.companySearch"
  results="modal.companies" result-label="result.name" model-options="modal.searchOptions"
  createable="true" createable-condition="!modal.exactCompanyFound(searchModel) && (modal.step !== 'new-company' || modal.companySearch !== modal.newCompany.name)"
  createable-label='Create company "{{modal.companySearch}}"' on-select="modal.selectCompany(result)"
  on-create="modal.createCompany(searchModel)"></div>
```

Longest line is at about 144.

When I set the wrap line length to 40, I get this:

```
 <div autocomplete placeholder="Type to select a company or create a new one*"
  class="search-container search-filter-container company-search"
  ng-class="{'for-new-company': modal.newCompany}"
  search-model="modal.companySearch" results="modal.companies"
  result-label="result.name" model-options="modal.searchOptions"
  createable="true" createable-condition="!modal.exactCompanyFound(searchModel) && (modal.step !== 'new-company' || modal.companySearch !== modal.newCompany.name)"
  createable-label='Create company "{{modal.companySearch}}"'
  on-select="modal.selectCompany(result)"
  on-create="modal.createCompany(searchModel)"></div>
```

Seems to me that the check for wrapping is only when the attribute BEGINS after the specified amount - I would expect an alternate mode where any attribute that ENDS after the length will try to cut off, if it's too long on its own, it should cut itself off and then be in its own line that is longer and will be wrapped.

So evidently, I would like this result:

```
<div autocomplete
  placeholder="Type to select a company or create a new one*"
  class="search-container search-filter-container company-search"
  ng-class="{'for-new-company': modal.newCompany}"
  search-model="modal.companySearch"
  results="modal.companies"
  result-label="result.name"
  model-options="modal.searchOptions"
  createable="true"
  createable-condition="!modal.exactCompanyFound(searchModel) && (modal.step !== 'new-company' || modal.companySearch !== modal.newCompany.name)"
  createable-label='Create company "{{modal.companySearch}}"'
  on-select="modal.selectCompany(result)"
  on-create="modal.createCompany(searchModel)"></div>
```

Which thinking of it, can be reduced to another type of behavior: I know you can set `"wrap_attributes"` to be `auto` or `force`. `force` is closest to what I want, but the problem is that it forces even small elements with 2-3 attributes, which I'd rather keep in one line. I'm thinking there should be a third option, that will work like `force`, but only in the case of over `x` (let's say 5?) attributes on the same element. 

Are one of these possible? Thank you!

**Comment by bitwiseman** [comment]
Thank you for a detailed issue report.
The wrapping behavior you would like to see seems much more correct than the current behavior, but the engine in that part of the project (html beautifier) may not support it yet.  I'll take a look.

**Comment by rodfersou** [comment]
I have the same problem here

**Comment by rodfersou** [comment]
I try to fix the method space_or_wrap .. can someone guide me if this is the right way to fix it or can contribute with a better idea?

**Comment by thedamon** [comment]
what if you could pass an integer to `wrap_attributes` and that would make it wrap when there are at least that number of attributes present? (or fewer if max line-length is reached.. though perhaps that should be a separate rule)

I don't think any of the current options mimic how most people in the wild would format HTML

**Comment by bitwiseman** [comment]
Fixed in 1.9.0.
